pkg/util/ranger/detacher.go

```diff
@@ -362,6 +362,26 @@ func unionColumnValues(lhs, rhs []*valueInfo) []*valueInfo {
 	return lhs
 }
 
+// Check which detach result is more selective. This function is called to choose between point ranges and the best CNF ranges.
+// This is needed because sometimes the best CNF has full intersection and is more selective,
+// and other times it is not when the intersection is not applied.
+func chooseBetweenRangeAndPoint(sctx *rangerctx.RangerContext, r1 *DetachRangeResult, r2 *cnfItemRangeResult) *DetachRangeResult {
+	if fixcontrol.GetBoolWithDefault(sctx.OptimizerFixControl, fixcontrol.Fix54337, false) {
+		if r1 != nil && len(r1.Ranges) > 0 && r2 != nil && r2.rangeResult != nil {
+			r1Minusr2 := removeConditions(sctx.ExprCtx.GetEvalCtx(), r1.AccessConds, r2.rangeResult.AccessConds)
+			r2Minusr1 := removeConditions(sctx.ExprCtx.GetEvalCtx(), r2.rangeResult.AccessConds, r1.AccessConds)
+			// r2 covers more access conditions therefore it is more selective than r1. Make sure r1 has access condition to start with.
```

Comment:
Why `removeConditions` can work for this purpose? Could you add an example as comments here?

Reply:
The code comment says that "r2 covers more access conditions therefore it is more selective than r1. Make sure r1 has access condition to start with." and added that r1-r2 is empty implies that r2 is a superset of r1 with an example.